oneapp-server/types/ApplicationContact.js

```diff
@@ -8,55 +8,55 @@ const typeDef = gql`
 
   type ApplicationContact {
     "Applicant first name"
-    APPLICANT_FIRST_NAME: String
+    APPLICANT_FIRST_NAME: String! @constraint(pattern: "^[a-zA-Z.''-'\s]*$", maxLength: 25)
     "Applicant last name"
-    APPLICANT_LAST_NAME: String
+    APPLICANT_LAST_NAME: String! @constraint(pattern: "^[a-zA-Z.''-'\s]*$", maxLength: 25)
     "Applicant middle name"
-    APPLICANT_MIDDLE_NAME: String
+    APPLICANT_MIDDLE_NAME: String @constraint(pattern: "^[a-zA-Z.''-'\s]*$", maxLength: 25)
     "Application maiden name"
-    APPLICANT_MAIDEN_NAME: String
+    APPLICANT_MAIDEN_NAME: String @constraint(pattern: "^[a-zA-Z.''-'\s]*$", maxLength: 25)
     "Payee first name, if different than applicant"
-    PAYEE_FIRST_NAME: String
+    PAYEE_FIRST_NAME: String! @constraint(pattern: "^[a-zA-Z.''-'\s]*$", maxLength: 25)
     "Payee last name, if different than applicant"
-    PAYEE_LAST_NAME: String
+    PAYEE_LAST_NAME: String! @constraint(pattern: "^[a-zA-Z.''-'\s]*$", maxLength: 25)
     "Is applicant homeless"
     IS_HOMELESS: Boolean
```

Comment:
I noticed in the ASPX looks like this field is a dropdown (https://github.com/newjersey/dhs-oneapp-legacy/blob/master/NJOneAppAuto/Forms/Application/ApplicationContact.aspx#L233). Does it end up getting converted to a boolean in the database somewhere?

Reply:
I think this is something we'll want to think through as we build that final call between the frontend and GraphQL. Not sure how much validation/conversion we'll want to do in the GraphQL mutation layer. My intuition is that we do a quick string to boolean conversion in the frontend, so that if/when there are any string changes on the frontend, we won't need to touch the GraphQL/backend

Reply:
Agreed. Keeping presentation logic out of the API layer is ideal.

I was more thinking about how the database is storing boolean values. It looks like some fields in the database are using a 1 byte varchar stored as `Y/N` and `IS_HOMELESS` appears to be one. I'm wondering if there is a straightforward way we can convert those without having to go field by field. Can we use / write some GraphQL middleware to have the API interface be a Boolean but the underlying data operates on a `Y/N` string?